libknet/handle.c

```diff
@@ -534,6 +534,16 @@ static void _stop_threads(knet_handle_t knet_h)
 	}
 }
 
+/*
+ * It is safe to call log_* macros before _init_locks is completed and
+ * before threads are started, as long as the calls do not contain
+ * information about host_id or link_id or any other dynamic data structures
+ * that can change at runtime.
+ *
+ * This "issue" really exists only inside knet_handle_new_ex, and the current
+ * calls are all safe.
```

Comment:
Before `knet_handle_new_ex()` finishes, no other thread has a chance of using the handle being created, right? In that case, `knet_handle_new_ex()` could use a non-locking variant of `log_msg()`, and we could (hopefully) get rid of the `lock_init_done` flag as well. What do you think?

Reply:
Another option is initializing this lock before doing anything else and not trying to log its eventual failure but returning immediately with the error.

Reply:
If that's true (and I haven't had chance to check it), then a non-locking version of log_msg sounds like the neatest solution - with precedents all over the place in other code. a __log_msg for internal use perhaps?